.azure-pipelines/templates/stages/deploy-stage.yml

```diff
@@ -59,3 +62,26 @@ stages:
                 snapcraft upload --release=${{ parameters.snapReleaseChannel }} "${SNAP_FILE}"
               done
             displayName: Publish to Snap store
+      - job: docker
+        pool:
+          vmImage: ubuntu-18.04
+        steps:
+          - task: Docker@2
+            inputs:
+              command: login
+              # The credentials used here are for the shared certbotbot account
+              # on Docker Hub. The credentials are stored in a service account
+              # which was created by following the instructions at
+              # https://docs.microsoft.com/en-us/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml#sep-docreg.
+              # The name given to this service account must match the value
+              # given to containerRegistry below. "Grant access to all
+              # pipelines" should also be checked.  To revoke these
+              # credentials, we can change the password on the certbotbot
+              # Docker Hub account or remove the account from the
+              # Certbot organization on Docker Hub.
+              containerRegistry: docker-hub
+            displayName: Login to Docker Hub
+          - bash: tools/docker/build.sh  ${{ parameters.dockerTag }}
```

Comment:
Do we need to have the build step at release time? We don't currently run it manually during a release.

Reply:
We do.

Currently the way things work in the `certbot-docker` repo is we run `deploy.sh` which pushes a new tag to that GitHub repo and we have Docker Hub configured to build and publish Docker images for us on new tags.

With this PR though, we're building and publishing the Docker image in Azure. Docker Hub no longer builds the images. It just hosts them. We're doing this to get more control over the process to give us things like nightly builds and fix https://github.com/certbot-docker/certbot-docker/issues/20.

Because Docker Hub isn't building the images though, we have to build them ourselves so that they can be published by the new `deploy.sh` script.

Reply:
Ah, gotcha. I was wondering where that extra building was happening.